gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/TraversalRing.java

```diff
@@ -62,6 +62,10 @@ public void addTraversal(final Traversal.Admin<A, B> traversal) {
         this.traversals.add(traversal);
     }
 
+    public void setTraversal(final int index, final Traversal.Admin<A, B> traversal) {
```

Comment:
Yea, it would be cleaner, but with indexing it's much faster as you don't have to search the whole `TraversalRing` for a specific traversal. I know, the ring will usually contain only a few traversals, but well.... It's `O(1)` vs. `O(n)`.